tbdex/quote/quote.go

```diff
@@ -0,0 +1,252 @@
+package quote
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/TBD54566975/tbdex-go/tbdex"
+	"github.com/tbd54566975/web5-go/dids/did"
+	"go.jetpack.io/typeid"
+)
+
+// Kind identifies this message kind
+const Kind = "quote"
+
+// Quote represents a quote message within the exchange.
+type Quote struct {
+	MessageMetadata tbdex.MessageMetadata `json:"metadata"`
+	Data            Data                  `json:"data"`
+	Signature       string                `json:"signature"`
+}
+
+// Data encapsulates the data content of a  quote.
+type Data struct {
+	ExpiresAt string       `json:"expiresAt"`
+	Payin     QuoteDetails `json:"payin"`
+	Payout    QuoteDetails `json:"payout"`
+}
+
+// QuoteDetails describes the relevant information of a currency that is being sent or received
+type QuoteDetails struct {
+	CurrencyCode       string              `json:"currencyCode"`
+	Amount             string              `json:"amount"`
+	Fee                string              `json:"fee,omitempty"`
+	PaymentInstruction *PaymentInstruction `json:"paymentInstruction,omitempty"`
+}
+
+// PaymentInstruction contains instructions with plain text and/or a link
+type PaymentInstruction struct {
+	Link        string `json:"link,omitempty"`
+	Instruction string `json:"instruction,omitempty"`
+}
+
+// Digest computes a hash of the quote
+func (q Quote) Digest() ([]byte, error) {
+	payload := map[string]any{"metadata": q.MessageMetadata, "data": q.Data}
+
+	hashed, err := tbdex.DigestJSON(payload)
+	if err != nil {
+		return nil, fmt.Errorf("failed to digest quote: %w", err)
+	}
+
+	return hashed, nil
+}
+
+// Sign cryptographically signs the quote using DID's private key
+func (q *Quote) Sign(bearerDID did.BearerDID) error {
+	q.MessageMetadata.From = bearerDID.URI
+
+	signature, err := tbdex.Sign(q, bearerDID)
+	if err != nil {
+		return fmt.Errorf("failed to sign rfq: %w", err)
+	}
+
+	q.Signature = signature
+
+	return nil
+}
+
+// Verify verifies the signature of the quote.
+func (q *Quote) Verify() error {
+	_, err := tbdex.VerifySignature(q, q.Signature)
+	if err != nil {
+		return fmt.Errorf("failed to verify quote signature: %w", err)
+	}
+
+	// TODO add check when decoded.SignerDID is implemented
+	// if decoded.SignerDID != r.MessageMetadata.From {
+	// 	return errors.New("signer: %w does not match message metadata from: %w", decoded.Header.SignerDID, r.MessageMetadata.From)
+	// }
```

Comment:
this is now possible in latest `web5-go` version

Reply:
I thiiink @jiyoontbd is addressing in https://github.com/TBD54566975/tbdex-go/pull/25?

Reply:
ah i'll just implement for this one

Reply:
i can do this work easy peasy

Reply:
done: https://github.com/TBD54566975/tbdex-go/pull/26/commits/588aa2b17d3525cfe2f2b7fe93ca16d5f18b83db